test/puppetlabs/puppetdb/testutils/db.clj

```diff
@@ -128,51 +135,62 @@
         (jdbc/do-commands-outside-txn
          "create extension if not exists pg_trgm"
          "create extension if not exists pgcrypto"))
-      (let [cfg (db-user-config template-name)]
+      (let [cfg (routine-db-config template-name)]
         (jdbc/with-db-connection cfg
-          (migrate!)))
+          (initialize-schema)))
       (reset! template-created true))))
 
+(defn- require-suitable-pg-arrangement
+  [{:keys [user migrator-username subname] :as config}]
+  (jdbc/with-db-connection config
+    (let [db (subname->validated-db-name subname)
+          priv? jdbc/has-database-privilege?
+          migrator migrator-username]
+      (when (priv? "public" db "connect")
+        (throw (Exception. "public connect")))
+      (when-not (priv? user db "connect")
+        (throw (Exception. "no user connnect")))
+      (when (priv? user db "connect with grant option")
```

Comment:
Should the user only have a connect without the grant option here? I thought the migrator was the only one that should have "connect with grant option"

Reply:
ah, never mind I read that wrong